src/test/java/com/fryrank/controller/ReviewControllerTests.java

```diff
@@ -0,0 +1,51 @@
+package com.fryrank.controller;
+
+import com.fryrank.dal.ReviewDAL;
+import com.fryrank.model.Review;
+import org.junit.Assert;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.junit.MockitoJUnitRunner;
+
+import java.util.ArrayList;
+import java.util.List;
+
+
+@RunWith(MockitoJUnitRunner.class)
+public class ReviewControllerTests {
+    @Mock
+    ReviewDAL reviewDAL;
+
+    @InjectMocks
+    ReviewController controller;
+
+    @Test
+    public void testGetAllReviewsForRestaurant() throws Exception {
+        List<Review> expectedReviews = new ArrayList<>() {
+                {
+                    add(new Review("review_id_1", "restaurant_id_1", "author_id_1", 5.0 , "title_1", "body_1"));
+                    add(new Review("review_id_2", "restaurant_id_2", "author_id_2", 7.0 , "title_2", "body_2"));
+                }
+        };
+        Mockito.when(reviewDAL.getAllReviewsByRestaurantId("1")).thenReturn(expectedReviews);
+
+        List<Review> actualReviews = controller.getAllReviewsForRestaurant("1");
+
+        Assert.assertEquals(actualReviews, expectedReviews);
+    }
+
+    @Test
+    public void testAddNewReviewForRestaurant() throws Exception {
+        Review expectedReview = new Review("review_id_1", "restaurant_id_1", "author_id_1", 5.0 , "title_1", "body_1");
+
+        Mockito.when(reviewDAL.addNewReview(expectedReview)).thenReturn(expectedReview);
+
+        Review actualReview = controller.addNewReviewForRestaurant(expectedReview);
+
+        Assert.assertEquals(expectedReview, actualReview);
+    }
+
```

Comment:
Can you add tests for null input cases? For example, restaurantId is null, review is null, and some attribute of the Review object is null

Reply:
I think this is a similar case to https://github.com/NickPriv/FryRankBackend/pull/47/files#r1500227397, where we're really only testing the code path.

Reply:
I think it's different because we have @NonNull annotations which technically have two conditions that can be tested, null or non null parameters being provided.

Reply:
Ah testing the annotation makes sense.

My above comment was more directed to restaurant ID and other attributes are null part.

Reply:
Done in d068ca1e463c3737dbe88c9adc9cfac2249427e6.

Reply:
Done in 6cbdb1d3f16e40ee7a0515c1bb07f5b4ee96e6c9.